pkg/nodeconfig/nodeconfig.go

```diff
@@ -346,19 +346,15 @@ func createKubeletConf(clusterServiceCIDR string) (string, error) {
 		return "", err
 	}
 	kubeletConfig := generateKubeletConfiguration(clusterDNS)
-	kubeletConfigData, err := json.Marshal(kubeletConfig)
+	kubeletConfigBytes, err := json.Marshal(kubeletConfig)
 	if err != nil {
 		return "", err
 	}
-	// Replace last character ('}') with comma
-	kubeletConfigData[len(kubeletConfigData)-1] = ','
-	// Appending this option is needed here instead of in the kubelet configuration object. Otherwise, when marshalling,
-	// the empty value will be omitted, so it would end up being incorrectly populated at service start time.
-	// Can be moved to kubelet configuration object with https://issues.redhat.com/browse/WINC-926
-	enforceNodeAllocatable := []byte("\"enforceNodeAllocatable\":[]}")
-	kubeletConfigData = append(kubeletConfigData, enforceNodeAllocatable...)
-
-	return string(kubeletConfigData), nil
+	kubeletConfigStrData := string(kubeletConfigBytes)
+	// Windows kubelet does not recognize the uppercase versions of these arguments
+	kubeletConfigStrData = strings.ReplaceAll(kubeletConfigStrData, "CgroupsPerQOS", "cgroupsPerQOS")
+	kubeletConfigStrData = strings.ReplaceAll(kubeletConfigStrData, "EnforceNodeAllocatable", "enforceNodeAllocatable")
```

Comment:
After doing these replacements the node is able to be configured/join the cluster but things like `oc adm node-logs` breaks. I suspect every JSON key in the config file needs to be replaced like this unless we fix the marshalling upstream.